10up/insert-special-characters
# add allow-licenses check to Dependency Review action

### Description of the Change
This builds on from #126 and further configures the [dependency review](https://github.blog/changelog/2022-06-15-configure-dependency-review-for-vulnerability-severity-and-license-type/) for license type (also feasible is the vulnerability severity with the `fail-on-severity` that I've currently left commented out).  Assuming this correctly scans dependencies in the code and only passes dependencies with a GPL-compatible license then that's what we want (and can then close #122, #133 as previous attempts with other tools that didn't quite work out).

### How to test the Change
I want to first see what happens when this runs against the entire repo here.  After that we can add the following two dependencies temporarily in this PR to see that the check fails and then subsequently passes.
1. Add https://www.npmjs.com/package/react-chevron and see that the action fails.
2. Remove the above and see that the action passes.
3. Add https://www.npmjs.com/package/apache-2.0-licensed and see that the action passes.

### Changelog Entry

> Added - GPL-compatible license check for dependencies.

### Credits

Props @jeffpaul, @vikrampm1.

### Checklist:
